### PR TITLE
Rewrite frontend for NC 28

### PR DESCRIPTION
**Pull request by julien-nc**
* Remove DocuSign/LibreSign related code
* Use Php8 syntax
* Rewrite frontend for NC 28
    * Replace the info sidebar elements by an inline file action (showing a modal when clicked)
    * Implement request/approve/reject file actions with `@nextcloud/files` 
    * Use `registerDavProperty` from `@nextcloud/files` to get approval-related props
    * Get inspiration from https://github.com/nextcloud/files_lock to dynamically update file list items
 * Remove deprecation warning
 * Remove compiled asset (if we agree on not shipping this app with releases anymore)

| Context | Before | After |
| --- | --- | --- |
| Current user can approve/reject a file | Inline indicator image In the files details sidebar image | Inline action image And this modal appears when clicking on the inline action image |
| Current user cannot approve/reject | Inline indicator image Details sidebar image | Inline action image Modal image |
| Approved file |Inline indicator image Details sidebar image | Inline action image Modal image |

The file actions:
image

### CODE REVIEW

**Comment by julien-nc** [comment]
@jancborchardt @nimishavijay What do you think about the new UX? (mainly the inline action instead of the icon on top of the file thumbnail and the modal instead of adding stuff in the details sidebar)

A few doubts:
* Should we make the text in the inline action shorter? `Waiting for you approval` is a bit long
* Any idea of a better icon for the `Request approval` button in the modal and for the request file action?
* Better icon for the "pending" yellow icon next to `Approval request by`?

**Comment by skjnldsv** [comment]
> The file actions:
> [image](https://private-user-images.githubusercontent.com/11291457/291288315-82daf90f-65c3-42d0-a361-7d8f0776f85d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NzQ1OTcsIm5iZiI6MTcwMjk3NDI5NywicGF0aCI6Ii8xMTI5MTQ1Ny8yOTEyODgzMTUtODJkYWY5MGYtNjVjMy00MmQwLWEzNjEtN2Q4ZjA3NzZmODVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDA4MjQ1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5OGUyNjRlMDlmODdhZDc3YmVhZTc3Y2E2ZjYzMmNhMzVkMGEzYTdiNTBjOTkxMDI5ZTRhOWVmNTk2N2QxZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mFkB9KNK11CD12KEgCwgdf9mE_JM0goKQkDuIxYe8xQ)


Couldn't we have the actions inline?
Instead of in the menu? :)

File: src/filesplugin.js
Comment (skjnldsv) [inline src/filesplugin.js:42]: Files actions should also theoretically be registered on init :)

File: src/filesplugin.js
Comment (skjnldsv) [inline src/filesplugin.js:76]:
```suggestion
```

File: src/filesplugin.js
Comment (skjnldsv) [reply src/filesplugin.js:42]: Also, for cleaner code, I suggest you create one js file per action, and import/register them all in the init file.
Will be much easier to navigate on which action is where and does what ;)

**Comment by julien-nc** [comment]
@skjnldsv Thanks for the review :blue_heart: 

> Couldn't we have the approve/reject actions inline?
Instead of in the menu? :)

I think it would make the file item a bit crowded (state action + approve action + reject action). Informing inline about the state is enough. Clicking on the state opens a modal in which users can approve/reject so IMO it's fine.

Files are now split to the maximum :grin: and reorganized.

**Comment by skjnldsv** [comment]
image

- You can still request another approval even if you haven't approved/rejected a file?
- I would put the approve and reject first if there is a pending request :)

File: src/files/actions/approveAction.js
Comment (skjnldsv) [inline src/files/actions/approveAction.js:20]: No await?

File: src/files/actions/approveAction.js
Comment (skjnldsv) [inline src/files/actions/approveAction.js:29]: This will makes the selection stay smart and keep relevant files in case some failed to be approved.

```suggestion
		const promises = nodes
			.filter(node => node.attributes['approval-state'] === states.APPROVABLE)
			.map(onApproveAction)
		const results = await Promise.allSettled(promises)
		return results.map(promise => promise.status === 'fulfilled')
```

File: src/files/actions/rejectAction.js
Comment (skjnldsv) [inline src/files/actions/rejectAction.js:30]: Same as `approveAction`

File: src/files/actions/rejectAction.js
Comment (skjnldsv) [inline src/files/actions/rejectAction.js:20]:
```suggestion
```

File: src/files/actions/rejectAction.js
Comment (skjnldsv) [inline src/files/actions/rejectAction.js:21]: await ?

File: src/files/actions/requestAction.js
Comment (skjnldsv) [inline src/files/actions/requestAction.js:22]: Await ?

**Review by skjnldsv — COMMENTED** [review]
Some more comments
image

- [ ] You can still request another approval even if you haven't approved/rejected a file?
- [ ]  I would put the approve and reject first if there is a pending request :)
- [ ] Also, on another note, why returning silent in actions? I think it's best to make sure the user knows the action succeeded (or not), silent actions are mostly here if the action have more context that the standard `action xxxxx failed` message (e.g `You don't have enough free space available` instead of `The action "copy" failed`)

File: src/files/actions/approveAction.js
Comment (julien-nc) [reply src/files/actions/approveAction.js]: As the error messages are handled somewhere else, I think it does not matter if we wait or not, we don't wanna know anything anymore once we called `onApproveAction`. 
I still rewrote everything with async/await and exec/execBatch are now waiting for all promises to settle.

**Comment by julien-nc** [comment]
@skjnldsv 

> silent actions are mostly here if the action have more context that the standard `action xxxxx failed` message (e.g `You don't have enough free space available` instead of `The action "copy" failed`)

That's the case here, we don't want the standard `The action "XXX" failed` message. Success and failure messages are dealt with by the helper functions.

> You can still request another approval even if you haven't approved/rejected a file?

Yes, you can request again with another rule.

**Comment by skjnldsv** [comment]
> That's the case here, we don't want the standard `The action "XXX" failed` message. Success and failure messages are dealt with by the helper functions.

ah! Sorry I missed that :see_no_evil: 
Somehow having some UI logic in the helper functions seemed incoherent to me, so I didn't check :see_no_evil:

File: src/files/actions/approveAction.js
Comment (skjnldsv) [inline src/files/actions/approveAction.js:29]: You're still missing the error handling.
If you approve a selection, and some items fail, you want to keep the failed elements selected.
You're losing that information here

File: src/files/actions/rejectAction.js
Comment (skjnldsv) [inline src/files/actions/rejectAction.js:29]: Same as `src/files/actions/approveAction.js`

File: src/files/actions/approveAction.js
Comment (skjnldsv) [reply src/files/actions/approveAction.js]: It does, the selection is smart enough to keep failed items for the user convenience. See new review

**Review by skjnldsv — CHANGES_REQUESTED** [review]

File: src/files/actions/approveAction.js
Comment (julien-nc) [reply src/files/actions/approveAction.js]: My goal was to avoid the standard showError/showSuccess message. I would like to keep the failed elements selected but avoid the messages because I already produce custom ones.

Anyway, error handling for multiple selection approve and reject is now better and we have the additional standard message. Fine by me. Or is there a way to avoid it?

Single selection (exec for approve, reject, request and inline) are still silent.

Comment (julien-nc) [reply src/files/actions/approveAction.js]: Ok, now no more notification for each file in multiple selection.

**Review by skjnldsv — APPROVED** [review]
Amazing work! :rocket:

**Pull request merged**
